Even faster charging? + Add and enable Adreno Idler

GamingDevilsCC opened the pull request:
tiny code fix

GamingDevilsCC committed:
Even faster charging?

mountaser commented:
i need to look into that first let me get the new g3 and try it then i will Merge it !

GamingDevilsCC committed:
Adreno Idler

commits that added:
https://bitbucket.org/arter97/android_kernel_samsung_msm8974/commits/22d0b352f54a00029cfa18c368e9d680e3aaf60a
https://bitbucket.org/arter97/android_kernel_samsung_msm8974/commits/28c79e6018fd761245303feb51b1ad684399290f
https://bitbucket.org/arter97/android_kernel_samsung_msm8974/commits/6a001e2993138a7e44371632ab623e07d78e1a8b
https://bitbucket.org/arter97/android_kernel_samsung_msm8974/commits/b75554aa37be5dce5a83974d362d58d4cb7991a6
https://bitbucket.org/arter97/android_kernel_samsung_msm8974/commits/72b75f62452486c482dfd009ad038b1fdaf6a941
https://bitbucket.org/arter97/android_kernel_samsung_msm8974/commits/423c55f7ef34f73c0913a70f5d1463d94b612de3
https://bitbucket.org/arter97/android_kernel_samsung_msm8974/commits/c9fd20d3185ca0594c0175a3d1c8a6164e3956c2
https://bitbucket.org/arter97/android_kernel_samsung_msm8974/commits/d66cd80d6ca422fc1398bb0c821409e82bc0ba0d

partial. just the power suspend mods.
https://bitbucket.org/arter97/android_kernel_samsung_msm8974/commits/adebe9f7080df129d4cd5128fdbab2aeeba223b4

plus some extra lines of code from Artx kernel.

It should ramp down the frequency to the lowest after about 200~500ms when system hits idle.

And you can turn Adreno idler on/off and fine-tune from /sys/module/adreno_idler/parameters.

Benchmark
I've tested with watching a movie and playing a game.

1. 15 GB 1080p 2.5 hrs movie w/ HW decoder - Airplane mode ON/Brightness 25%
1.1. Adreno idler off - 46.3% battery drain, can expect approx. 6hrs 5mins of video playback on 100% battery charge.
1.2. Adreno idler on - 25.5% battery drain, can expect approx. 11hrs 2mins of video playback on 100% battery charge.

2. Casual game, leave it for 30 minutes on the same screen - Airplane mode ON/Brightness 100%
2.1. Adreno idler off - 13.4% battery drain, can expect approx, 3hrs 43mins of gameplay on 100% battery charge.
2.2. Adreno idler on - 11.7% battery drain, can expect approx, 4hrs 16mins of gameplay on 100% battery charge.

Source:
http://forum.xda-developers.com/android/software-hacking/adreno-idler-idling-algorithm-devfreq-t3134872

Signed-off-by: arter97 <qkrwngud825@gmail.com>

GamingDevilsCC committed:
Enable Adreno Idler

GamingDevilsCC committed:
Update lineageos_dualsim_defconfig
